src/main.ts

```diff
@@ -11,15 +11,19 @@ import {
 
 import { Controller } from './controller';
 import { IFileService } from '@core/interfaces/file-service.interface';
-import { PLATFORMS } from '@core/constants/main.constants';
+import { MacFilesService } from './services/mac-files.service';
 
-const isOSWindows = () => process.platform === PLATFORMS.WINDOWS;
+const getOS = () => process.platform;
+
+const OSService = {
+  linux: () => new LinuxFilesService(streamService),
+  win32: () => new WindowsFilesService(streamService),
+  darwin: () => new MacFilesService(streamService),
+};
```

Comment:
As an alternative to avoid the closure, you can do this:

```suggestion
const OSService = {
  linux: LinuxFilesService,
  win32: WindowsFilesService,
  darwin: MacFilesService,
};
```

Reply:
Done in 4145810